src/main/scala/mesosphere/marathon/core/task/tracker/impl/steps/PostToEventStreamStepImpl.scala

```diff
@@ -38,22 +38,34 @@ class PostToEventStreamStepImpl @Inject() (
       case TASK_ERROR | TASK_FAILED | TASK_FINISHED | TASK_KILLED | TASK_LOST =>
         // Remove from our internal list
         maybeTask match {
-          case Some(task) => postEvent(appId, status, task)
-          case None       => log.warn(s"Task not found. Do not post event for '{}'", taskId.getValue)
+          case Some(task) =>
+            postEvent(timestamp, appId, status, task)
+          case None =>
+            log.warn(s"Received terminal task event but task not found. " +
```

Comment:
Refactored it a bit.

In the short future, that logging statement becomes defacto unreachable but it's probably ok to leave it in there for robustness.